src/appendvec.rs

```diff
@@ -0,0 +1,261 @@
+use byteorder::{ByteOrder, LittleEndian};
+use memmap::MmapMut;
+use solana_sdk::account::Account;
+use solana_sdk::pubkey::Pubkey;
+use std::fs::{File, OpenOptions};
+use std::io::{Error, ErrorKind, Result, Seek, SeekFrom, Write};
+use std::marker::PhantomData;
+use std::mem;
+use std::path::Path;
+
+const SIZEOF_U64: u64 = mem::size_of::<u64>() as u64;
+
+pub struct AppendVec<T> {
+    data: File,
+    map: MmapMut,
+    current_offset: u64,
+    file_size: u64,
+    inc_size: u64,
+    phantom: PhantomData<T>,
+}
+
+impl<T> AppendVec<T>
+where
+    T: Default,
+{
+    pub fn new(path: &Path, create: bool, size: u64, inc: u64) -> Self {
+        let mut data = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(create)
+            .open(path)
+            .expect("Unable to open data file");
+
+        data.seek(SeekFrom::Start(size)).unwrap();
+        data.write_all(&[0]).unwrap();
+        data.seek(SeekFrom::Start(0)).unwrap();
+        data.flush().unwrap();
+        let map = unsafe { MmapMut::map_mut(&data).expect("failed to map the data file") };
+
+        AppendVec {
+            data,
+            map,
+            current_offset: 0,
+            file_size: size,
+            inc_size: inc,
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn get(&self, index: u64) -> T {
+        assert!(self.current_offset > index);
+        let index = (index as usize) * mem::size_of::<T>();
+        let data = &self.map[index..(index + mem::size_of::<T>())];
+        let x: T = unsafe { std::ptr::read(data.as_ptr() as *const _) };
```

Comment:
i think this is a copy.  I couldn't get atomics to work over the T returned for the same index.

https://github.com/aeyakovenko/appendvec/blob/master/src/appendvec.rs#L46

in my change get returns a reference to the memory.  That seems to work.